LibreNMS/Device/Sensor.php

```diff
@@ -424,7 +429,7 @@ protected static function processSensorData($sensors, $prefetch)
                 $sensor_value = (cast_number($sensor_value) * $sensor['sensor_multiplier']);
             }
 
-            $sensor_data[$sensor['sensor_id']] = $sensor_value;
+            $sensor_data[$sensor['sensor_id'] ?? 0] = $sensor_value; // if no sensor_id, this is a new sensor, place at the first index
```

Comment:
Since it's in a loop, it could overwrite index 0 multiple times?

Reply:
Yeah, I had a hard time understanding this code.   The only time sensor_id is null is during discovery of a new (unsaved) sensor.
After fetching the data the discovery runs current($data) (Arr::first essentially) so it only takes one result out of the array.

Reply:
The code needs to be tossed, but it is currently working :/